### PR TITLE
Small updates to Http response/request and Udp

### PR DESCRIPTION
**Pull request by slaff**
- added methods to control multicast Time-To-Live(TTL) and multicast IP
- made the toString() methods in HttpRequest and HttpResponse to be available also in SMING_RELEASE mode.

### CODE REVIEW

File: Sming/Core/Network/Http/HttpRequest.cpp [hunk 1]
```diff
@@ -83,20 +83,20 @@ void HttpRequest::reset()
...
+	content += String(http_method_str(method)) + ' ' + uri.getPathWithQuery() + _F(" HTTP/1.1\r\n");
 	content += headers.toString(HTTP_HEADER_HOST, uri.getHostWithPort());
 	for(unsigned i = 0; i < headers.count(); i++) {
 		content += headers[i];
 	}
 
 	if(bodyStream != nullptr && bodyStream->available() >= 0) {
 		content += headers.toString(HTTP_HEADER_CONTENT_LENGTH, String(bodyStream->available()));
+	} else {
+		content += _F("\r\n");
```
Comment (mikee47) [inline Sming/Core/Network/Http/HttpRequest.cpp:98]: Don't bother with _F, "\r\n" is very commonly used so linker will merge all instances.

File: Sming/Core/Network/Http/HttpRequest.h [hunk 2]
```diff
@@ -248,13 +248,11 @@ class HttpRequest
 		return this;
 	}
 
-#ifndef SMING_RELEASE
 	/**
 	 * @brief Tries to present a readable version of the current request values
 	 * @retval String
 	 */
 	String toString();
```
Comment (mikee47) [inline Sming/Core/Network/Http/HttpRequest.h:255]: Should also add a `String toString(const HttpRequest& req)` function for consistency with other APIs.

File: Sming/Core/Network/Http/HttpResponse.cpp [hunk 3]
```diff
@@ -140,6 +140,23 @@ void HttpResponse::reset()
...
+	String content;
+	content += F("HTTP/1.1 ") + String(code) + F(" ") + httpGetStatusText(code) + F(" \r\n");
+	for(unsigned i = 0; i < headers.count(); i++) {
+		content += headers[i];
+	}
+
+	if(stream != nullptr && stream->available() >= 0) {
+		content += headers.toString(HTTP_HEADER_CONTENT_LENGTH, String(stream->available()));
+	} else {
+		content += F("\r\n");
```
Comment (mikee47) [inline Sming/Core/Network/Http/HttpResponse.cpp:154]: Just "\r\n", F() not necessary.

**Review by mikee47 — CHANGES_REQUESTED** [review]
I've used some igmp_joingroup amd igmp_leavegroup functions in SSDP Server.cpp, could you perhaps add those as well?

**Review by mikee47 — APPROVED** [review]

**Comment by slaff** [comment]
@mikee47 Thanks for the fixes. Shall I merge also this PR or you have some more changes in the back of your mind?

**Pull request merged**